src/utils/format.ts

```diff
@@ -24,8 +24,7 @@ const FORMAT = {
 /* -------------------------------------------------------------------------- */
 
 export const toBN = (amount?: Amount | number): BigNumber => {
-  if (!amount || amount === '') amount = '0';
-  return new BigNumber(amount);
+  return new BigNumber(amount || 0);
```

Comment:
are we sure we want to return a bn of 0 instead of throwing an error here? @xgambitox  @karelianpie

Reply:
What the actual `BigNumber` from 'bignumber.js' does for example in case its given a `null` is
```typescript
new BigNumber(null) // BigNumber { s: null, e: null, c: null }
```
I think big number of 0 is sensible for us, but I don't have a strong opinion here. The change above was just a refactor, the logic remained as it was, just saving a line since `amount || 0` is the same as what the other line was doing. I'm happy to refactor to throw if you Turtle or Gambito have a strong opinion either way.

Reply:
Yes its intentional. We save a lot of lines of code in many places where amount might be undefined, and then we dont need to place safeguards to continue with the math operation.